### PR TITLE
Implemented share links for VKontakte and Odnoklassniki

### PR DESCRIPTION
**Pull request by marinovdf**
Fix [PrestaShop/PrestaShop#17886](https://github.com/PrestaShop/PrestaShop/issues/17886)

Additionally fixed the fork project name in the README.md.

### CODE REVIEW

**Comment by marinovdf** [comment]
Also need to add the icons to base theme

**Comment by marinovdf** [comment]
What about PR?

**Comment by matks** [comment]
I'm sorry but I'm not sure these social networks as are popular as the others 🤔 

@PrestaShop/prestashop-product-team wdyt ?

**Comment by ttoine** [comment]
VKontacte is the leader in Russia, that would makes sense to have it for this market/area.

Eventually, if this is not possible to add new one "per default", is this possible to create a way to add custom social networks with a form? it would require to manage 2 fields: name (text), picture/logo (file upload). Some CMS allows to do that, this is very useful: it can also be used to place a link to a RSS, to a blog, or to a marketplace, or other custom ways to follow the shop.

File: ps_sharebuttons.php [hunk 1]
```diff
@@ -206,6 +208,22 @@ public function getWidgetVariables($hookName, array $params)
             );
         }
 
+        if (Configuration::get('PS_SC_VKONTAKTE')) {
+            $social_share_links['vkontakte'] = array(
+                'label' => $this->trans('VKontakte', array(), 'Modules.Sharebuttons.Shop'),
+                'class' => 'vkontakte',
+                'url' => 'http://vk.com/share.php?url='.$sharing_url,
```
Comment (sam-pires) [inline ps_sharebuttons.php]: Might need to use https instead of http here

**Comment by sam-pires** [comment]
Hi, 

Having the possibility to share on VK an OK.ru is great. 
VK monthly traffic =  1.87B
OK.ru monthly traffic = 892.08M

**One question, how does it work for the icon in the front-end?**


---
On another note, I agree with @ttoine, to avoid ending up with a list of 100 social media, we should add the possibility to add custom social media at some point. We'll look into creating some specs for that.

**Comment by sam-pires** [comment]
I created an issue to link this PR to, I add the link in the Description.

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
You need to resolve conflicts before continue the review process

**Comment by Progi1984** [comment]
Hello @marinovdf, thanks for your contribution, would you mind to rebase your pull request? Thanks

File: ps_sharebuttons.php (hunk 1 above)
Comment (marinovdf) [reply ps_sharebuttons.php]: Sure. Fixed

**Review by marinovdf — COMMENTED** [review]
Hello @Progi1984. I've make rebase.

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by sarahdib** [comment]
@marinovdf Thank you for this contribution the PR is ok 


@PrestaShop/prestashop-designers we should integrate the icon for the new social media what do you think ?
Capture d’écran 2020-05-22 à 10 11 51

**Comment by sam-pires** [comment]
Definitely needs the icons :)

**Comment by Progi1984** [comment]
@NeOMakinG It seems that icons from this module use these classes for loading icons (https://github.com/PrestaShop/PrestaShop/blob/develop/themes/classic/_dev/css/components/footer.scss#L65). Could you help @marinovdf and improve for having icons to this PR ? Thanks

**Comment by NeOMakinG** [comment]
You just need to add svg icons into the img folder as it is done for others social networks, and then add classes there like this is done already : https://github.com/PrestaShop/PrestaShop/blob/develop/themes/classic/_dev/css/components/footer.scss#L65

**Comment by ttoine** [comment]
@LouiseBonnard please note that SVG vector graphic pictures are used in this case already.

**Comment by Progi1984** [comment]
@NeOMakinG So this plugin runs only with the theme classic and not independently. Yes ?

**Comment by NeOMakinG** [comment]
@Progi1984 didn't think about it but it looks like yes, damn this is weird

**Comment by prestascott** [comment]
Hi, 

I agree and we should definitely add VK and Ok.ru icons otherwise share buttons remain blank and cannot be identified as these social networks. Since we're using google material for icons and fontawesome to complete the missing ones, i recommend using those svg for the icons:
- VK: https://fontawesome.com/icons/vk?style=brands
- Ok.ru: https://fontawesome.com/icons/odnoklassniki?style=brands

**Comment by Progi1984** [comment]
@PrestaShop/prestashop-product-team Does this module have to work independently? Or Can we add icons to `classic` theme ?

**Comment by sam-pires** [comment]
How does it work for existing social network icons?

**Comment by Progi1984** [comment]
@sam-pires It fetchs icon from the `classic` theme from specific CSS class.

**Comment by sam-pires** [comment]
and it uses a specific list of icons in the theme or it uses fontawesome?

**Comment by Progi1984** [comment]
from this theme

**Comment by sam-pires** [comment]
Ok so after discussion, let's add the icons to the `classic` theme as it has been done with other social networks.

**Comment by Progi1984** [comment]
Ping @NeOMakinG

**Comment by LouiseBonnard** [comment]
> Ok so after discussion, let's add the icons to the `classic` theme as it has been done with other social networks.

Could we know why? 🙏

**Comment by sam-pires** [comment]
> Could we know why? 🙏

you can ;)

There are 2 options:
- Adding the icons to the `classic` theme, making the module dependant on the theme
- Making icons manageable at the module level

The module has always worked the first way. So making it work differently means modify how it works for all the social networks already available in the module.

For the moment let's keep it that way. The PR has been opened for a long time, let's not change everything now. 


Nevertheless, it is interesting to make it work differently, allowing the users to add their own social networks in the module. So everyone is free to add the ones they like, and it doesn't require PrestaShop to add and update all social networks needed by the users.

I created a 'feedback' about it in Productboard already. We could create an issue if you want, but we already know a lot about the needs for this module.

**Comment by colinegin** [comment]
ping @Progi1984 and @NeOMakinG what is the status of this PR ? Can it go to QA ?

**Comment by NeOMakinG** [comment]
> Hi,
> 
> I agree and we should definitely add VK and Ok.ru icons otherwise share buttons remain blank and cannot be identified as these social networks. Since we're using google material for icons and fontawesome to complete the missing ones, i recommend using those svg for the icons:
> 
> * VK: https://fontawesome.com/icons/vk?style=brands
> * Ok.ru: https://fontawesome.com/icons/odnoklassniki?style=brands

Someone should add these icons to the classic theme in order to continue with this PR

@marinovdf do you want to do it ?

Or I'll just do it

**Comment by matks** [comment]
Hi everybody

Since we had no news from the author for more than 6 months, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Feel free to reopen or open another one if you think it's still relevant 😉 

Thanks!

**Pull request closed without merging**